core/chain_indexer.go

```diff
@@ -203,7 +203,7 @@ func (c *ChainIndexer) eventLoop(currentHeader *types.Header, events chan ChainE
 			if header.ParentHash != prevHash {
 				// Reorg to the common ancestor (might not exist in light sync mode, skip reorg then)
 				// TODO(karalabe, zsfelfoldi): This seems a bit brittle, can we detect this case explicitly?
-				if h := FindCommonAncestor(c.chainDb, prevHeader, header); h != nil {
+				if h := FindCommonAncestor(c.chainDb, prevHeader, header); h != nil && h.Hash() != prevHash {
```

Comment:
I'm really afraid of tiny changes like this which have no explanation and seem out of place. Please provide a test that repros a failure this means to fix.

Reply:
It does not cause an actual failure even if it unnecessarily detects a reorg. I am thinking about a test that explicitly tests reorg detection but since this is not urgent at all and does not need to get into 1.8.0 I decided to remove it from this PR. I will probably add it to https://github.com/ethereum/go-ethereum/pull/15378 instead which deals with the same logic and is also not urgent, just a cosmetic change.